apps/pg_subscriber/lib/pg_subscriber/tuple_data.ex

```diff
@@ -1,29 +1,41 @@
 defmodule PgSubscriber.TupleData do
-
   alias __MODULE__
   alias PgSubscriber.Column
 
   @type t :: %__MODULE__{
-    num_of_cols: integer,
-    columns: [Column.t()],
-  }
+          num_of_cols: integer,
+          columns: [Column.t()]
+        }
 
   defstruct [:num_of_cols, :columns]
 
   @spec get_tuple_data(binary()) ::
-    {:ok, TupleData.t(), binary()}
-    | {:error, term()}
+          {:ok, TupleData.t(), binary()}
+          | {:error, term()}
   @doc """
   Parses a TupleData binary message and returns the parsed TupleData struct and the remaining binary.
   In case of invalid format, an error with reason is returned.
   """
   def get_tuple_data(data) do
     with <<num_of_cols::16, rest::binary>> <- data,
-    {:ok, columns, rest} <- Column.get_cols(num_of_cols, rest) do
+         {:ok, columns, rest} <- Column.get_cols(num_of_cols, rest) do
       {:ok, %TupleData{num_of_cols: num_of_cols, columns: columns}, rest}
     else
       {:error, reason} -> {:error, reason}
       _ -> {:error, {:invalid_data_length, data}}
     end
   end
+
+  @spec convert_to_map(TupleData.t() | nil) :: map()
```

Comment:
I agree that we should have `TupleData` or similar in `core`, I was thinking the same.
But do we really want to have it represented as `map`? As you mentioned in TODO, it would only make sense if we had column name as well, but A) we don't, and B) if I understand it correctly the whole row (tuple) is sent for every update (even if only one column was updated for example), so I would say that we should represent it as `tuple` or `list`.